drivers/mhz19/mhz19_uart.c

```diff
@@ -0,0 +1,256 @@
+/*
+ * Copyright (C) 2018 Koen Zandberg <koen@bergzand.net>
+ * Copyright (C) 2018 Beduino Master Projekt - University of Bremen
+ * Copyright (C) 2020 Bas Stottelaar <basstottelaar@gmail.com>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_mhz19
+ * @{
+ *
+ * @file
+ * @brief       Device driver implementation for MH-Z19 CO2 sensor.
+ *
+ * @author      Koen Zandberg <koen@bergzand.net>
+ * @author      Christian Manal <manal@uni-bremen.de>
+ * @author      Bas Stottelaar <basstottelaar@gmail.com>
+ *
+ * @}
+ */
+
+#include "mhz19.h"
+#include "mhz19_params.h"
+#include "xtimer.h"
+#include "mutex.h"
+
+#define ENABLE_DEBUG        (0)
+#include "debug.h"
+
+#ifdef MODULE_MHZ19_UART
+#include "mhz19_internals.h"
+
+/* Precalculated command sequences */
+static const uint8_t value_read[] = {
+    MHZ19_READ_START,
+    MHZ19_READ_SENSOR_NUM,
+    MHZ19_CMD_GAS_CONCENTRATION,
+    0x00,
+    0x00,
+    0x00,
+    0x00,
+    0x00,
+    0x79 /* checksum */
+};
+
+static const uint8_t auto_calibration_on[] = {
+    MHZ19_READ_START,
+    MHZ19_READ_SENSOR_NUM,
+    MHZ19_CMD_AUTO_CALIBRATION,
+    0xA0, /* on */
+    0x00,
+    0x00,
+    0x00,
+    0x00,
+    0xE6 /* checksum */
+};
+
+static const uint8_t auto_calibration_off[] = {
+    MHZ19_READ_START,
+    MHZ19_READ_SENSOR_NUM,
+    MHZ19_CMD_AUTO_CALIBRATION,
+    0x00, /* off */
+    0x00,
+    0x00,
+    0x00,
+    0x00,
+    0x86 /* checksum */
+};
+
+static const uint8_t calibrate_zero_point[] = {
+    MHZ19_READ_START,
+    MHZ19_READ_SENSOR_NUM,
+    MHZ19_CMD_CALIBRATE_ZERO,
+    0x00,
+    0x00,
+    0x00,
+    0x00,
+    0x00,
+    0x78 /* checksum */
+};
+
+static void _mhz19_timeout(void *arg)
+{
+    mhz19_t *dev = arg;
+
+    mutex_unlock(&(dev->sync));
+}
+
+static void _mhz19_rx_cb(void *arg, uint8_t byte)
+{
+    mhz19_t *dev = arg;
+
+    /* Skip start byte and skip out of array bounds writes */
+    if ((dev->idx == 0 && byte == 0xff) || dev->idx >= MHZ19_BUF_SIZE) {
+        return;
+    }
+    /* Store byte and increment idx */
+    dev->rxmem[dev->idx++] = byte;
+    if (dev->idx == MHZ19_BUF_SIZE) {
+        mutex_unlock(&dev->sync);
+    }
+}
+
+int mhz19_init(mhz19_t *dev, const mhz19_params_t *params)
+{
+    dev->params = params;
+
+    DEBUG("mhz19: initializing device %p on UART %i\n",
+          (void *)dev, dev->params->uart);
+
+    mutex_init(&dev->mutex);
+    mutex_init(&dev->sync);
```

Comment:
```suggestion
    mutex_init_locked(&dev->sync);
```

Reply:
`mutex_init_locked` does not exist, so it's `mutex_init` followed by a `mutex_lock`.